lib/api/src/modules/stories.ts

```diff
@@ -262,38 +262,47 @@ export const init: ModuleFn = ({
 
       const kindSlug = storyId?.split('--', 2)[0];
 
-      if (!story) {
-        const s = kindOrId ? hash[kindOrId] || hash[sanitize(kindOrId)] : hash[kindSlug];
-        // eslint-disable-next-line no-nested-ternary
-        const id = s ? (s.children ? s.children[0] : s.id) : kindOrId;
-        let viewMode =
-          s && !isRoot(s) && (viewModeFromArgs || s.parameters.viewMode)
-            ? s.parameters.viewMode
-            : viewModeFromState;
-
-        // Some viewModes are not story-specific, and we should reset viewMode
-        //  to 'story' if one of those is active when navigating to another story
-        if (['settings', 'about', 'release'].includes(viewMode)) {
-          viewMode = 'story';
+      if (!name) {
+        // Find the entry (group, component or story) that is referred to
+        let entry = titleOrId ? hash[titleOrId] || hash[sanitize(titleOrId)] : hash[kindSlug];
+
+        if (!entry) {
+          throw new Error(`Unknown id or title: '${titleOrId}'`);
         }
 
-        const p = s && s.refId ? `/${viewMode}/${s.refId}_${id}` : `/${viewMode}/${id}`;
+        // We want to navigate to the first ancestor entry that is a leaf
+        while (!entry.isLeaf) entry = hash[entry.children[0]];
+
+        // We would default to the viewMode passed in or maintain the current
+        const desiredViewMode = viewModeFromArgs || viewModeFromState;
+
+        // By default we would render a story as a story
+        let viewMode = 'story';
+        // However, any story can be rendered as docs if required
+        if (desiredViewMode === 'docs') {
+          viewMode = 'docs';
+        }
+        // If the entry is actually a component, we are looking at docs also
```

Comment:
How does this work?

Reply:
Currently my data format for a "docs only" entry is a component that is also a leaf--ie has no children. That may be a bad way to mode it though. 

What do you think?

Reply:
Sounds dicey, but maybe fine for now?